### PR TITLE
BIA-1185: Implement API to AMT parquet conversion - core collection - StudentProgramDim

### PR DESCRIPTION
**Pull request by jegasape**
- Implement API to AMT parquet conversion - core collection - StudentProgramDim

### CODE REVIEW

**Comment by DavidJGapCR** [comment]
`StudentProgramDim` belongs to Core collection, not Equity collection.

**Comment by DavidJGapCR** [comment]
Core collection is in the base folder. Core collection and Base collection are the same one.

File: src/edfi_amt_data_lake/parquet/amt/base/base_collection.py [hunk 1]
```diff
@@ -23,4 +26,5 @@ def base_collection(school_year) -> None:
     local_education_agency_dim(school_year)
     most_recent_grading_period(school_year)
     school_dim(school_year)
+    student_program_dim(school_dim)
```
Comment (DavidJGapCR) [inline src/edfi_amt_data_lake/parquet/amt/base/base_collection.py:29]: `student_program_dim(school_year)`

File: src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        'schoolReference.schoolId',
+        'programReference.programName',
+        'programReference.programTypeDescriptor',
+        'programReference.educationOrganizationId',
+        'beginDate',
+        'programId',
+        'graduationPlanReference.educationOrganizationId',
+    ])
+
+    data_frame['LastModifiedDate'] = ''
```
Comment (DavidJGapCR) [inline src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py:91]: You can remove this line. This column is not necessary.

File: src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    data_frame['beginDate'] = to_datetime_key(data_frame, 'beginDate')
+
+    data_frame['StudentSchoolProgramKey'] = (
+        data_frame['studentReference.studentUniqueId'].astype(str) + '-'
+        + data_frame['schoolReference.schoolId'].astype(str) + '-'
+        + data_frame['programReference.programName'].astype(str) + '-'
+        + data_frame['programId'].astype(str) + '-'
+        + data_frame['programReference.educationOrganizationId'].astype(str) + '-'
+        + data_frame['programReference.educationOrganizationId'].astype(str) + '-'
```
Comment (DavidJGapCR) [inline src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py:101]: Same column twice in the concatenation?

File: src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    data_frame['LastModifiedDate'] = ''
+
+    data_frame['beginDate'] = to_datetime_key(data_frame, 'beginDate')
+
+    data_frame['StudentSchoolProgramKey'] = (
+        data_frame['studentReference.studentUniqueId'].astype(str) + '-'
+        + data_frame['schoolReference.schoolId'].astype(str) + '-'
+        + data_frame['programReference.programName'].astype(str) + '-'
+        + data_frame['programId'].astype(str) + '-'
```
Comment (DavidJGapCR) [inline src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py:99]: Here you need to include the program type descriptor instead. You have 2 options: 
1. merge to the descriptor using the namespace and the code value. 
2. Use the value from this descriptor column as it is.

File: src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py (hunk 3 above)
Comment (DavidJGapCR) [reply src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py:101]: Notice that the `StudentProgramAssociations` has `educationOrganizationReference.educationOrganizationId` and `programReference.educationOrganizationId`

Comment (jegasape) [reply src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py:97]: Done.

File: src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py (hunk 2 above)
Comment (jegasape) [reply src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py]: Done.

File: src/edfi_amt_data_lake/parquet/amt/base/base_collection.py (hunk 1 above)
Comment (jegasape) [reply src/edfi_amt_data_lake/parquet/amt/base/base_collection.py]: Done.

File: src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py (hunk 4 above)
Comment (jegasape) [reply src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py]: Done.

File: src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        'programReference.educationOrganizationId',
+        'programReference.programTypeDescriptor',
+        'studentReference.studentUniqueId',
+        'schoolReference.schoolId',
+        'graduationPlanReference.educationOrganizationId',
+    ])
+
+    data_frame['beginDate'] = to_datetime_key(data_frame, 'beginDate')
+
+    data_frame['StudentSchoolProgramKey'] = (
```
Comment (DavidJGapCR) [inline src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py:92]: We are missing the program name in this concatenation. That means you have to add it to the subset above as well.

**Review by DavidJGapCR — APPROVED** [review]

**Pull request merged**